src/raster/gcp.rs

```diff
@@ -1,8 +1,111 @@
 //! Raster ground control point support
 
+use std::ffi::{CStr, CString};
+use std::marker::PhantomData;
+
+use gdal_sys::CPLErr;
+
+use crate::errors::Result;
 use crate::spatial_ref::SpatialRef;
+use crate::utils::{_last_cpl_err, _string};
 use crate::Dataset;
 
+/// An owned Ground Control Point.
+#[derive(Clone, Debug, PartialEq, PartialOrd)]
+pub struct Gcp {
+    /// Unique identifier, often numeric.
+    pub id: String,
+    /// Informational message or empty string.
+    pub info: String,
+    /// Pixel (x) location of GCP on raster.
+    pub pixel: f64,
+    /// Line (y) location of GCP on raster.
+    pub line: f64,
+    /// X position of GCP in georeferenced space.
+    pub x: f64,
+    /// Y position of GCP in georeferenced space.
+    pub y: f64,
+    /// Elevation of GCP, or zero if not known.
+    pub z: f64,
+}
+
+/// A wrapper over a Ground Control Point, borrowed from an existing dataset.
+#[derive(Clone, Debug)]
+#[repr(transparent)]
+pub struct GcpRef<'a> {
```

Comment:
Is the lifetime parameter necessary? `inner` will be valid even if the originating `Dataset` goes away.

Reply:
It won't, it's the dataset's memory IIRC.

Reply:
Ahhh... because there's a cast happening here:

https://github.com/georust/gdal/pull/404/files#diff-19ab0c93d77643f78caf7ad8ff89235da2a204d84593581fa208cb70d51caca7R133

Thank you.